### PR TITLE
Archive the `gallina` tool

### PR DESCRIPTION
**Pull request by vbgl**
The `gallina` utility is a complete and stand-alone software, whose [source](https://github.com/coq/coq/blob/V8.8.0/tools/gallina_lexer.mll) [code](https://github.com/coq/coq/blob/V8.8.0/tools/gallina.ml) and [documentation](https://github.com/coq/coq/blob/V8.8.0/man/gallina.1) are publicly accessible online.

Keeping its source within the master branch of this repository does not seem to bring anything but burned resources: bandwidth, CI time and energy, `coq_makefile` rules, …

Also, various people use the name “Gallina” to refer to some language of Coq, not to this legacy utility. Keeping these few files around might foster some confusion.

- [x] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [x] Entry added in CHANGES.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -41,6 +41,8 @@ Tools
   COQFLAGS is now entirely separate from COQLIBS, so in custom Makefiles
   $(COQFLAGS) should be replaced by $(COQFLAGS) $(COQLIBS).
 
+- Remove gallina.
```
Comment (Zimmi48) [inline CHANGES]: You were talking about confusion, you should definitely expand this CHANGES entry to avoid some.

**Comment by Zimmi48** [comment]
As mentioned in other PRs removing stuff from the main Coq repo, if the tool is deemed useful, it is always possible to consider keeping maintaining in the scope of https://github.com/coq-community (this only requires one motivated user to become the maintainer).

File: CHANGES (hunk 1 above)
Comment (vbgl) [reply CHANGES]: Good catch.

**Review by gares — APPROVED** [review]

**Comment by ejgallego** [comment]
12 days and no asignee, there are like 3 owners so I won't step in, please, do assign.

**Review by maximedenes — APPROVED** [review]

**Pull request merged**